interactors/transaction.go

```diff
@@ -0,0 +1,164 @@
+package interactors
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/core"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/data"
+)
+
+var log = logger.GetOrCreate("interactors")
+
+// ArgCreateTransaction will hold the transaction fields
+type ArgCreateTransaction struct {
+	Nonce     uint64
+	Value     string
+	RcvAddr   string
+	SndAddr   string
+	GasPrice  uint64
+	GasLimit  uint64
+	Data      []byte
+	Signature string
+	ChainID   string
+	Version   uint32
+	Options   uint32
+}
+
+type transactionInteractor struct {
+	Proxy
+	TxSigner
+	mutTxAccumulator           sync.RWMutex
+	millisecondsBetweenBunches uint32
+	txAccumulator              []*data.Transaction
+}
+
+// NewTransactionInteractor will create an interactor that extends the proxy functionality with some transaction-oriented functionality
+func NewTransactionInteractor(proxy Proxy, txSigner TxSigner) (*transactionInteractor, error) {
+	if check.IfNil(proxy) {
+		return nil, ErrNilProxy
+	}
+	if check.IfNil(txSigner) {
+		return nil, ErrNilTxSigner
+	}
+
+	return &transactionInteractor{
+		Proxy:    proxy,
+		TxSigner: txSigner,
+	}, nil
+}
+
+func (ti *transactionInteractor) SetMillisecondsBetweenBunches(millisecondsBetweenBunches uint32) {
```

Comment:
add a comment to exported functions.
Also, please set a default value for `ti.millisecondsBetweenBunches` on the constructor. if this setter is not called, then there will be no sleeping period